microsoft/testsuites/core/azure_image_standard.py

```diff
@@ -1115,32 +1115,51 @@ def verify_no_pre_exist_users(self, node: Node) -> None:
         ),
     )
     def verify_resource_disk_readme_file(self, node: RemoteNode) -> None:
-        resource_disk_mount_point = node.features[Disk].get_resource_disk_mount_point()
-
-        # verify that resource disk is mounted
-        # function returns successfully if disk matching mount point is present
-        node.features[Disk].get_partition_with_mount_point(resource_disk_mount_point)
-
-        # Verify lost+found folder exists
-        # Skip this step for BSD as it does not have lost+found folder
-        # since it uses UFS file system
-        if not isinstance(node.os, BSD):
-            fold_path = f"{resource_disk_mount_point}/lost+found"
-            folder_exists = node.tools[Ls].path_exists(fold_path, sudo=True)
-            assert_that(folder_exists, f"{fold_path} should be present").is_true()
-
-        # verify DATALOSS_WARNING_README.txt file exists
-        file_path = f"{resource_disk_mount_point}/DATALOSS_WARNING_README.txt"
-        file_exists = node.tools[Ls].path_exists(file_path, sudo=True)
-        assert_that(file_exists, f"{file_path} should be present").is_true()
-
-        # verify 'WARNING: THIS IS A TEMPORARY DISK' contained in
-        # DATALOSS_WARNING_README.txt file.
-        read_text = node.tools[Cat].read(file_path, force_run=True, sudo=True)
-        assert_that(
-            read_text,
-            f"'WARNING: THIS IS A TEMPORARY DISK' should be present in {file_path}",
-        ).contains("WARNING: THIS IS A TEMPORARY DISK")
+        node_disc = node.features[Disk]
+        resource_disks = node_disc.get_resource_disks()
```

Comment:
Add a method in `Disk` feature, called like `get_disk_controller_type`, and it accept mount point as a parameter. Please also reduce duplicate code with existing `get_os_disk_controller_type`.

Reply:
Fixed it.

Reply:
Please refine `get_os_disk_controller_type` to use the same logic. You can see the `get_os_disk_controller_type` contains logic for FreeBSD too. If you implement another `get_resource_disk_type`. The FreeBSD needs to fix the new failure again.

Reply:
Added new method "get_disc_type" to re-use in both get_os_disk_controller_type and get_resource_disk_type

Reply:
Please use `disk` instead of `disc` to keep consistent.

Reply:
Fixed